File: internal/model/maintenance_alert.go
package model

import "time"

type MaintenanceAlert struct {
	ID           int       `db:"id_alerta" json:"id_alerta"`
	Type         string    `db:"tipo_alerta" json:"tipo_alerta"`
	Date         time.Time `db:"fecha_alerta" json:"fecha_alerta"`
	KMAlert      int       `db:"km_alerta" json:"km_alerta,omitempty"`
	Status       string    `db:"estado_alerta" json:"estado_alerta"`
	Message      string    `db:"mensaje" json:"mensaje,omitempty"`
	Seen         bool      `db:"visto" json:"visto"`
	ConfigID     int       `db:"id_conf_mant" json:"id_conf_mant"`
}

File: src/main/java/com/maintenancesystem/maintenanceSystem/entity/MaintenanceAlert.java
package com.maintenancesystem.maintenanceSystem.entity;

import com.maintenancesystem.maintenanceSystem.enums.*;
import jakarta.persistence.*;
import lombok.*;
import java.time.LocalDate;

@Entity
@Table(name = "alerta_mantenimiento")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class MaintenanceAlert {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id_alerta")
    private Integer idAlert;

    @Enumerated(EnumType.STRING)
    @Column(name = "tipo_alerta", columnDefinition = "ENUM('preventiva','correctiva') DEFAULT 'preventiva'")
    private AlertType alertType;

    @Column(name = "fecha_alerta", nullable = false)
    private LocalDate alertDate;

    @Column(name = "km_alerta")
    private Integer kmAlert;

    @Enumerated(EnumType.STRING)
    @Column(name = "estado_alerta", columnDefinition = "ENUM('notificada','atendida','vencida') DEFAULT 'notificada'")
    private AlertStatus alertStatus;

    @Column(name = "mensaje", columnDefinition = "TEXT")
    private String message;

    @Column(name = "visto", columnDefinition = "BOOLEAN DEFAULT FALSE")
    private Boolean viewed = false;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_conf_mant", nullable = true, foreignKey = @ForeignKey(name = "fk_alerta_configuracion"))
    private MaintenanceConfiguration maintenanceConfiguration;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_vehiculo")
    private Vehicle vehicle;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "id_tipo_mant")
    private MaintenanceType maintenanceType;

}
